tools/clang/test/HLSLFileCheck/hlsl/intrinsics/mul/mul.hlsl

```diff
@@ -0,0 +1,992 @@
+// This file contains tests covering all overloads of mul intrinsic
+// as documented here: https://docs.microsoft.com/en-us/windows/win32/direct3dhlsl/dx-graphics-hlsl-mul
+
+// TODO: While testing overloads of mul() intrinsics, found some incorrect codegen for bool type.
+// Add coverage for bool type once the issue #2467 is fixed.
+
+// *****************************
+// float overloads
+// *****************************
+
+// vectors and scalars
+// RUN: %dxc -T vs_6_0 -E main -DELEM_TY1=float4 -DELEM_TY2=float4 -DRET_TY=float %s  | FileCheck %s -check-prefix=FL4_OVRLD
```

Comment:
I didn't realize we allowed RUN lines anywhere in the file, neat!

How did you generate/verify all the CHECK lines?  I'm not sure I have the patience to verify all of these by hand during CR...

Reply:
Yeah, while enabling support for multiple RUN lines, I noticed that FileChecker would always expect the RUN line at the beginning of the file, so removed that restriction.

I wrote the test manually for floats and then replicated for other types. It was a bit tedious. :/

Reply:
Thanks for all the effort!  BTW: I think part of the purpose of RUN lines having to be first (other than test performance or false positives) was to make it clear (to human readers) what variants of the test were going to be run by just looking at the top of the file, instead of having to look through the whole file for all the RUN lines.  I'm not sure we want to make the change to support run lines anywhere in the file.

Reply:
okay. I can add back the restriction for RUN lines in a separate pr.

Reply:
If we wanted, as a compromise, due to the benefit of having run lines next to their associated CHECK lines in cases like this, we could have a special RUN line that indicates that the real run lines are to be found throughout the file, rather than all grouped at the beginning.  Perhaps something like `// RUN-LINES-DEFERRED`, or `// RUN-LOCATED-ANYWHERE`, or `// RUN-LINES-SCATTERED`.  I kind of like the last one.  But this feature is something we could discuss and add if we find it valuable.